### PR TITLE
Updates based on a review in advance of the November 2018 IETF meeting.

### PR DESCRIPTION
**Pull request by jmillican**
- Remove "join an existing group" from architecture doc, based on UserAdd having been removed from the protocol doc.
- Remove statement that message delivery cannot be correlated between AS identities and DS delivery channels. This seems generally very difficult to achieve. It may be possible at times, but does not seem to be grounded in discussions that we have had at the working group meetings.
- Note that federation requires compatible wire encodings.
- Note that communication with the AS requires TLS or alternative authentication.
- Note that out-of-band channels may facilitate detection of misbehaviour. 
- Some grammar fixes.

### CODE REVIEW

File: draft-ietf-mls-architecture.md [hunk 1]
```diff
@@ -205,8 +205,6 @@ Clients may wish to do the following:
 
  -  remove one or more members from an existing group;
 
- -  join an existing group;
```
Comment (raphaelrobert) [inline draft-ietf-mls-architecture.md]: I suppose this relates to UserAdd having been removed from the protocol draft?
If so, the idea was to re-introduce it later, so I'm wondering if this requirement shouldn't stay in the architecture draft in the meantime (as a goal for the protocol)

File: draft-ietf-mls-architecture.md [hunk 2]
```diff
@@ -561,7 +559,8 @@ weakening the PCS guarantees for attachments.
 
 In general we do not consider Denial of Service (DoS) resistance to be the responsibility
 of the protocol. However, it should not be possible for anyone to perform a
-trivial DoS attack from which it is hard to recover.
+trivial DoS attack from which it is hard to recover. This excludes the DS, which
```
Comment (raphaelrobert) [inline draft-ietf-mls-architecture.md]: "This excludes the DS" -> "This excludes the DS as an attacker"

File: draft-ietf-mls-architecture.md (hunk 1 above)
Comment (beurdouche) [reply draft-ietf-mls-architecture.md]: I agree with Raphael, this should be kept.

File: draft-ietf-mls-architecture.md [hunk 3]
```diff
@@ -429,16 +427,15 @@ are made about the format of the payload.
 
 The protocol is designed in a way that limits the server-side (AS and DS)
 metadata footprint. The DS only persists data required for the delivery
-of messages and avoid Personally Identifiable Information (PII) or other
```
Comment (beurdouche) [inline draft-ietf-mls-architecture.md]: Please revert this change, this is not an editorial decision. Moreover I disagree with removing it except if we have consensus that we cannot achieve it.

File: draft-ietf-mls-architecture.md [hunk 4]
```diff
@@ -463,7 +460,8 @@ goals ("MLS aims to guarantee that...")?]]
 We assume that all transport connections are secured via some transport
 layer security mechanism such as TLS {{?I-D.ietf-tls-tls13}}. However,
 as noted above, the security of MLS will generally survive compromise
-of the transport layer.
+of the transport layer. Note that this may not extend to connections with the
+AS, which will typically require authentication at a minimum.
```
Comment (beurdouche) [inline draft-ietf-mls-architecture.md]: What do you mean here ?

File: draft-ietf-mls-architecture.md (hunk 2 above)
Comment (beurdouche) [reply draft-ietf-mls-architecture.md]: You can revert this change. "Anyone" is already contains the DS.

File: draft-ietf-mls-architecture.md [hunk 5]
```diff
@@ -607,9 +606,9 @@ including total DoS attacks (where it simply refuses to forward any
 messages) and partial DoS attacks (where it refuses to forward
 messages to and from specific Clients). As noted in
 {{delivery-guarantees}}, these attacks are only partially detectable
-by clients. Ultimately, failure of the DS to provide reasonable
-service must be dealt with as a customer service matter, not via
-technology.
+by clients without an out-of-band channel. Ultimately, failure of
```
Comment (beurdouche) [inline draft-ietf-mls-architecture.md]: What do you mean here ?

File: draft-ietf-mls-architecture.md (hunk 2 above)
Comment (jmillican) [reply draft-ietf-mls-architecture.md]: Oh sorry @beurdouche, I meant that the DS is able to DoS, as opposed to anyone else who shouldn't be able to. Will update to @raphaelrobert's suggestion.

File: draft-ietf-mls-architecture.md (hunk 3 above)
Comment (jmillican) [reply draft-ietf-mls-architecture.md]: Fair enough :) I wanted to include as a point for discussion, as it seems a fundamentally unrealistic goal to achieve. Let's discuss on the list.

File: draft-ietf-mls-architecture.md (hunk 4 above)
Comment (jmillican) [reply draft-ietf-mls-architecture.md]: We claim the MLS should be secure even without any sort of transport security. This makes sense for the DS, but if we are to have an AS which authoritatively authenticates identities, this will have to be authenticated.

File: draft-ietf-mls-architecture.md (hunk 5 above)
Comment (jmillican) [reply draft-ietf-mls-architecture.md]: It feels like too strong an assertion to claim that there can be no technical solution to detecting a targeted DoS by the DS. Something out of band might work; such as a transparency protocol.
